File: mobile/packages/expo-ghostty-terminal/android/src/main/java/expo/modules/ghosttyterminal/GhosttyTerminalView.kt
package expo.modules.ghosttyterminal

import android.content.Context
import android.graphics.Color
import expo.modules.kotlin.AppContext
import expo.modules.kotlin.viewevent.EventDispatcher
import expo.modules.kotlin.views.ExpoView

// Why: this view will host Ghostty natively; the react-native-webview terminal is
// being cut off for energy reasons — do NOT implement full terminal rendering here yet.
class GhosttyTerminalView(context: Context, appContext: AppContext) : ExpoView(context, appContext) {
    val onReady by EventDispatcher()
    val onTerminalInput by EventDispatcher()
    val onQueryReply by EventDispatcher()

    init {
        // Placeholder backing; the Ghostty surface will replace this later.
        setBackgroundColor(Color.BLACK)
    }
}

File: mobile/packages/expo-ghostty-terminal/ios/GhosttyTerminalView.swift
import ExpoModulesCore

// Why: this view will host Ghostty natively; the react-native-webview terminal is
// being cut off for energy reasons — do NOT implement full terminal rendering here yet.
public final class GhosttyTerminalView: ExpoView {
    let onReady = EventDispatcher()
    let onTerminalInput = EventDispatcher()
    let onQueryReply = EventDispatcher()

    required init(appContext: AppContext? = nil) {
        super.init(appContext: appContext)
        // Placeholder backing; the Ghostty surface will replace this later.
        backgroundColor = .black
    }
}
